### PR TITLE
deps/pip: Update `rules_python` -> 0.26.0 (+add pip-compile)

### PR DESCRIPTION
**Pull request by phlax**
This finally seems to work - previous versions of `rules_python` failed with our deps for one reason/another

This adds a ci check to ensure the requirements in `tools/base` are in sync

It also adds a bazel target for updating:

```console
$ bazel run //tools/base:requirements.update
```

This doesnt update to the latest for all packages, just ensures the requirements.in reqs are met by current reqs.txt

There is a chance that this will create some unwanted change due to paths and dependabot - but i have a feeling this has been ~resolved

Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -735,9 +735,9 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Python rules for Bazel",
         project_desc = "Bazel rules for the Python language",
         project_url = "https://github.com/bazelbuild/rules_python",
-        version = "0.7.0",
-        sha256 = "15f84594af9da06750ceb878abbf129241421e3abbd6e36893041188db67f2fb",
-        release_date = "2022-03-11",
+        version = "b4d4b6d9a5d32a199b510ca3cd3273966ebaf987",
```
Comment (moderation) [inline bazel/repository_locations.bzl:738]: Can you use the latest release `0.8.1` - https://github.com/bazelbuild/rules_python/releases/tag/0.8.1

Comment (phlax) [reply bazel/repository_locations.bzl:738]: no - its using my pr/fork atm - it will take a bit more to get this working

Comment (phlax) [reply bazel/repository_locations.bzl:738]: looks like my pr is about to land - will update this when it is and we can see where its at

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**